components/org.wso2.carbon.config/src/main/java/org/wso2/carbon/config/ConfigurationUtils.java

```diff
@@ -159,4 +159,27 @@ public static String getSystemVariableValue(String variableName, String defaultV
         }
         return value;
     }
+
+    /**
+     * Returns replace value with escaped characters.
+     * @param value replaced values
+     * @return value with escaped characters
+     */
+    public static String escapeSpecialCharacters(String value) {
+        if (value == null) {
+            return null;
+        }
+        StringBuilder sb = new StringBuilder();
+        for (int i = 0; i < value.length(); i++) {
+            char c = value.charAt(i);
+            if (c == '\\' || c == '+' || c == '-' || c == '!'  || c == '(' || c == ')' || c == ':'
```

Comment:
Is it possible to simplify this statement. IMO we could use an array of these chars and check if char `c` is contained in the array. WDYT?